msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -1500,58 +1498,58 @@ private AccountRecord selectAccountRecordForTokenRequest(
             // If the account is named like <tenant_name>.onmicrosoft.com we need to query the OpenId
             // Provider Configuration Metadata in order to get the tenant id. Once we have the
             // tenant id, we must then select the appropriate home or profile.
-            String tenantIdNameOrAlias = aadAuthority.getAudience().getTenantId();
+            String tenantId = aadAuthority.getAudience().getTenantId();
 
             // The AccountRecord we'll use to request a token...
             final AccountRecord accountRecord = new AccountRecord();
             accountRecord.setEnvironment(multiTenantAccount.getEnvironment());
             accountRecord.setHomeAccountId(multiTenantAccount.getHomeAccountId());
 
-            final boolean isUuid = isUuid(tenantIdNameOrAlias);
+            final boolean isUuid = isUuid(tenantId);
 
-            if (!isUuid && !isHomeTenantEquivalent(tenantIdNameOrAlias)) {
-                final OpenIdProviderConfiguration providerConfiguration =
-                        loadOpenIdProviderConfigurationMetadata(requestAuthority);
-
-                final String issuer = providerConfiguration.getIssuer();
-                final Uri issuerUri = Uri.parse(issuer);
-                final List<String> paths = issuerUri.getPathSegments();
-
-                if (paths.isEmpty()) {
-                    final String errMsg = "OpenId Metadata did not contain a path to the tenant";
-
-                    com.microsoft.identity.common.internal.logging.Logger.error(
-                            TAG,
-                            errMsg,
-                            null
-                    );
-
-                    throw new MsalClientException(errMsg);
-                }
-
-                tenantIdNameOrAlias = paths.get(0);
+            if (!isUuid && !isHomeTenantEquivalent(tenantId)) {
```

Comment:
nit: I personally found the name "isHomeTenantEquivalent" sort of confusing. Not sure what "common", "organization", "consumers" should be called though.